### PR TITLE
feat: add bundle_size job in CCI flow

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

-adds bundle_size job in CCI flow
-upgrades node version in build executor

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jimblanc — APPROVED** [review]
Nice, good first step! Could you confirm that you got a clean integ test run (pretty big cypress upgrade)? Also we should hold off on merging to main until we discuss this afternoon

**Comment by israx** [comment]
> Could you confirm that you got a clean integ test run (pretty big cypress upgrade)? I think the integration tests are running on another executor that has a different node version.

File: .circleci/config.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ version: 2.1
 executors:
   build-executor:
     docker:
-      - image: cypress/base:12.22.8
+      - image: cypress/base:14.15.0
```
Comment (haverchuck) [inline .circleci/config.yml:8]: Have we verified that all other tests run successfully with this version bump?

File: packages/auth/package.json [hunk 2]
```diff
@@ -28,8 +28,15 @@
...
     "clean": "rimraf lib-esm lib dist",
     "format": "echo \"Not implemented\"",
-    "lint": "tslint '{__tests__,src}/**/*.ts'"
+    "lint": "tslint '{__tests__,src}/**/*.ts'",
+    "bundle-size": "size-limit"
   },
+   "size-limit": [
+       {
+        "limit": "41.633 kB",
+        "path": "dist/aws-amplify-auth.min.js"
```
Comment (haverchuck) [inline packages/auth/package.json:37]: This is testing the size of the minified file only.  If that's what we are going to do for now, maybe we should change the command names to "size-test" or something, and not use the term "bundle".

Have you looked into testing the actual impact of the auth/core modules on a web app bundled with webpack?

File: package.json [hunk 3]
```diff
@@ -85,6 +87,7 @@
 		"rollup-plugin-node-resolve": "^4.0.0",
 		"rollup-plugin-sourcemaps": "^0.4.2",
 		"rollup-plugin-typescript": "^1.0.0",
+		"size-limit": "^8.1.0",
```
Comment (haverchuck) [inline package.json:90]: This is a CLI tool, yeah? Have other teams looked at this and vetted it with security?

**Review by haverchuck — CHANGES_REQUESTED** [review]
See inline comments.

File: package.json (hunk 3 above)
Comment (israx) [reply package.json:90]: The UI is using the same dev dependency

File: .circleci/config.yml (hunk 1 above)
Comment (israx) [reply .circleci/config.yml:8]: Unit tests run and work but need to verify if integration tests are using this version bump

Comment (iartemiev) [reply .circleci/config.yml:8]: Integ tests use different executors altogether, so they won't be affected

File: packages/auth/package.json (hunk 2 above)
Comment (iartemiev) [reply packages/auth/package.json:37]: +1

**Comment by abdallahshaban557** [comment]
Closing this - confirmed with @jimblanc

**Pull request closed without merging**